src/api/procedures/modifySignerPermissions.ts

```diff
@@ -34,7 +35,10 @@ export async function prepareModifySignerPermissions(
 
   const { secondaryAccounts } = args;
   const accounts = secondaryAccounts.map(({ account }) => account);
-  const existingSecondaryAccounts = await identity.getSecondaryAccountPermissions({ accounts });
+  const existingSecondaryAccounts = await getSecondaryAccountPermissions(
+    { accounts, identity },
+    context
+  );
   assertSecondaryAccounts(accounts, existingSecondaryAccounts);
```

Comment:
I think this function can be simplified to just checking if the lengths of both arrays are the same

Reply:
While that would work and give us consistent behavior with what we have, it seems like it would be a weird function to have. If it was only checking length then it should be called something like `assertSameLength`. Which seems too simple to me.

What about dropping it entirely and doing the check in line in the procedures instead?

Reply:
What about the `data` field in the error? I think we need this function if we want to tell the user which one is missing, where the length compare isn't going to do that. For now I am leaning towards leaving it in.

I think there is a simpler way to write the comparison inside the function though.

Reply:
With a functional style I have:

```typescript
  const addresses = accounts.map(({ address }) => address);
  const secondaryAddresses = secondaryAccounts.map(({ account: { address } }) => address);
  const notInTheList = addresses.filter(address => !secondaryAddresses.includes(address));
```